### PR TITLE
 [RHPAM-3330-str] Operator/Image support for Kie-Server Cluster support 

### PR DESCRIPTION
**Pull request by desmax74**
PR to test console-cr-form with jbpm as a string in a dropdown list 
https://issues.redhat.com/browse/KIECLOUD-507

### CODE REVIEW

File: pkg/apis/app/v2/kieapp_types.go [hunk 1]
```diff
@@ -252,6 +252,8 @@ type KieAppObject struct {
 	ImageTag string `json:"imageTag,omitempty"`
 	// The storageClassName to use
 	StorageClassName string `json:"storageClassName,omitempty"`
+	// Enable the clustering between Kieservers and add one replicas if only one is configured
+	JbpmCluster string `json:"jbpmCluster,omitempty"`
```
Comment (tchughesiv) [inline pkg/apis/app/v2/kieapp_types.go:256]: shouldn't this be a bool instead?

File: config/7.11.0/common.yaml [hunk 2]
```diff
@@ -680,6 +680,8 @@ servers:
                       value: "[[.Build.DisablePullDeps]]"
                     - name: KIE_SERVER_DISABLE_KC_VERIFICATION
                       value: "[[.Build.DisableKCVerification]]"
+                    - name: JBPM_CLUSTER
+                      value: "true"
```
Comment (tchughesiv) [inline config/7.11.0/common.yaml:684]: if this is set statically to true, how would the user change to false? maybe this needs to be a template var instead?
`[[.JbpmCluster]]` ?

File: pkg/controller/kieapp/defaults/defaults.go [hunk 3]
```diff
@@ -695,6 +696,19 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 			}
 
 			// Set replicas
+			// if the configuration use only one replica we increment
+			if len(cr.Spec.Objects.Servers) != 0 {
```
Comment (tchughesiv) [inline pkg/controller/kieapp/defaults/defaults.go:700]: at this point in the code, in this servers loop, we already know len is not 0. are we sure this check is necessary?

**Review by tchughesiv — CHANGES_REQUESTED** [review]

**Comment by desmax74** [comment]
@tchughesiv thanks for the review but this was just for a test with swati

**Pull request closed without merging**